src/ui_elements/appearance/color_picker.gd

```diff
@@ -11,19 +11,36 @@ var colorMapImage: StreamTexture
 var selectedColor: Color
 var selectedColorPosition: Vector2
 
+# --Public Functions--
+
+# Takes a position, scales it and displays the preview box
+func showPreview(pos: Vector2) -> void:
+	call_deferred("_showPreview", pos) # Call deferred runs it when it is ready
+
+# Returns the image scale of the color picker
+func getImageScale() -> Vector2:
+	var output = call_deferred("_getImageScale") # Call deferred runs it when it is ready
+	return(output)
```

Comment:
does the call_deferred work actually with the return value? I wonder how does this work.
Also, please make sure this runs if the window is resized.

Reply:
the scale is determined from _draw(), so it should always work no matter the window shape or size.

Reply:
I deleted the getImageScale anyways because it wasn't really useful and didn't work.